server/repository/src/db_diesel/period/period.rs

```diff
@@ -74,6 +74,8 @@ impl<'a> PeriodRepository<'a> {
                     apply_sort!(query, sort, period::end_date)
                 }
             }
+        } else {
+            query = query.order(period::end_date.desc());
```

Comment:
Will make a PR to have this paginate and use autocomplete with pagination. Don't think we need to load all the periods especially since some can be years old

Reply:
https://github.com/msupply-foundation/open-msupply/pull/6428#discussion_r1951544074